src/bootstrap/src/core/config/config.rs

```diff
@@ -2320,27 +2320,29 @@ impl Config {
                 Some(commit) => {
                     self.download_ci_rustc(commit);
 
-                    let builder_config_path =
-                        self.out.join(self.build.triple).join("ci-rustc/builder-config");
-                    let ci_config_toml = Self::get_toml(&builder_config_path);
-                    let current_config_toml = Self::get_toml(self.config.as_ref().unwrap());
-
-                    // Check the config compatibility
-                    let res = check_incompatible_options_for_ci_rustc(
-                        current_config_toml,
-                        ci_config_toml,
-                    );
+                    if let Some(config_path) = &self.config {
```

Comment:
Could we add a `// FIXME` here that points out that we should ideally deal with the actual config struct here (which will include `--set`), and not just the TOML file? I will try to refactor config parsing in the future so that it is possible, it will also be needed for using the new bootstrap command API in `Config::parse`.